src/main.c

```diff
@@ -1162,8 +1162,11 @@ static struct listener_config_t *add_listener(int fd, struct sockaddr *addr, soc
 {
     struct listener_config_t *listener = h2o_mem_alloc(sizeof(*listener));
 
+    memset(listener, 0, sizeof(*listener));
     memcpy(&listener->addr, addr, addrlen);
-    listener->fd = fd;
+    assert(listener->fds.size == 0);
+    h2o_vector_reserve(NULL, &listener->fds, 1);
+    listener->fds.entries[listener->fds.size++] = fd;
```

Comment:
I think that:
* We should initialize the properties in the order they are defined, rather than doing `fds` after `addr` (note: this is an issue with existing code but maybe we can fix this now).
* We might not need `assert(listener->fds.size == 0)` to test that `memset` does what it should?